src/components/ProfitAndLossReport/ProfitAndLossReport.tsx

```diff
@@ -0,0 +1,137 @@
+import { View } from '../View'
+import { ProfitAndLoss } from '../ProfitAndLoss'
+import { Panel } from '../Panel'
+import React, { RefObject, useContext, useState } from 'react'
+import { ProfitAndLossCompareOptionsProps } from '../ProfitAndLossCompareOptions/ProfitAndLossCompareOptions'
+import { ReportsStringOverrides } from '../../views/Reports/Reports'
+import { DateRangeDatePickerModes } from '../DatePicker/DatePicker'
+import { MoneyFormat } from '../../types'
+import { useLayerContext } from '../../contexts/LayerContext'
+import { Layer } from '../../api/layer'
+import { Button, ButtonVariant, RetryButton } from '../Button'
+import DownloadCloud from '../../icons/DownloadCloud'
+
+
+export interface ProfitAndLossReportProps {
+  stringOverrides?: ReportsStringOverrides
+  comparisonConfig?: ProfitAndLossCompareOptionsProps
+  datePickerMode?: DateRangeDatePickerModes
+  csvMoneyFormat?: MoneyFormat
+  parentRef?: RefObject<HTMLDivElement>
+}
+
+export interface DownloadButtonStringOverrides {
+  downloadButtonText?: string
+  retryButtonText?: string
+}
+
+export interface DownloadButtonProps {
+  stringOverrides?: DownloadButtonStringOverrides
+  moneyFormat?: MoneyFormat
+}
+
+export const PnLDownloadButton = ({
+  stringOverrides,
+  moneyFormat,
+}: DownloadButtonProps) => {
+  const { dateRange } = useContext(ProfitAndLoss.Context)
+  const { auth, businessId, apiUrl } = useLayerContext()
+  const [requestFailed, setRequestFailed] = useState(false)
+
+  const handleClick = async () => {
+    const month = (dateRange.startDate.getMonth() + 1).toString()
+    const year = dateRange.startDate.getFullYear().toString()
+    const getProfitAndLossCsv = Layer.getProfitAndLossCsv(
+      apiUrl,
+      auth.access_token,
+      {
+        params: {
+          businessId: businessId,
+          year: year,
+          month: month,
+          moneyFormat: moneyFormat,
+        },
+      },
+    )
+    try {
+      const result = await getProfitAndLossCsv()
+      if (result?.data?.presignedUrl) {
+        window.location.href = result.data.presignedUrl
+        setRequestFailed(false)
+      } else {
+        setRequestFailed(true)
+      }
+    } catch (e) {
+      setRequestFailed(true)
+    }
+  }
+
+  return requestFailed ? (
+    <RetryButton
+      onClick={handleClick}
+      className='Layer__download-retry-btn'
+      error={'Approval failed. Check connection and retry in few seconds.'}
+    >
+      {stringOverrides?.retryButtonText || 'Retry'}
+    </RetryButton>
+  ) : (
+    <Button
+      variant={ButtonVariant.secondary}
+      rightIcon={<DownloadCloud size={12} />}
+      onClick={handleClick}
+    >
+      {stringOverrides?.downloadButtonText || 'Download'}
+    </Button>
+  )
```

Comment:
We recently added new `<DownloadButton />` component that simplifies above and handles also `loading` state. New implementation already on `main` branch.

```suggestion
  return (
    <DownloadButtonComponent
      iconOnly={view === 'mobile'}
      onClick={handleClick}
      isDownloading={isDownloading}
      requestFailed={requestFailed}
      text={stringOverrides?.downloadButtonText || 'Download'}
      retryText={stringOverrides?.retryButtonText || 'Retry'}
    />
  )
```

...or you can try replace whole `PnLDownloadButton` with this:

```ts
import { DownloadButton as DownloadButtonComponent } from '../../components/Button'

const DownloadButton = ({
  stringOverrides,
  moneyFormat,
  view,
}: DownloadButtonProps) => {
  const { dateRange } = useContext(ProfitAndLoss.Context)
  const { auth, businessId, apiUrl } = useLayerContext()
  const [requestFailed, setRequestFailed] = useState(false)
  const [isDownloading, setIsDownloading] = useState(false)

  const handleClick = async () => {
    setIsDownloading(true)
    const month = (dateRange.startDate.getMonth() + 1).toString()
    const year = dateRange.startDate.getFullYear().toString()
    const getProfitAndLossCsv = Layer.getProfitAndLossCsv(
      apiUrl,
      auth.access_token,
      {
        params: {
          businessId: businessId,
          year: year,
          month: month,
          moneyFormat: moneyFormat,
        },
      },
    )
    try {
      const result = await getProfitAndLossCsv()
      if (result?.data?.presignedUrl) {
        window.location.href = result.data.presignedUrl
        setRequestFailed(false)
      } else {
        setRequestFailed(true)
      }
    } catch (e) {
      setRequestFailed(true)
    } finally {
      setIsDownloading(false)
    }
  }

  return (
    <DownloadButtonComponent
      iconOnly={view === 'mobile'}
      onClick={handleClick}
      isDownloading={isDownloading}
      requestFailed={requestFailed}
      text={stringOverrides?.downloadButtonText || 'Download'}
      retryText={stringOverrides?.retryButtonText || 'Retry'}
    />
  )
}
```